modules/openapi-generator/src/main/resources/python/configuration.mustache

```diff
@@ -26,11 +30,59 @@ class Configuration(object):
+    :param key_id: The identifier of the cryptographic key, when signing HTTP requests.
+        An 'Authorization' header is calculated by creating a hash of select headers,
+        and optionally the body of the HTTP request, then signing the hash value using
+        a private key which is available to the client.
+    :param private_key_path: The path of the file containing a private key,
+        when signing HTTP requests.
+    :param signing_scheme: The signature scheme, when signing HTTP requests.
+        Supported value is hs2019.
+    :param signing_algorithm: The signature algorithm, when signing HTTP requests.
+        Supported values are PKCS1v15, PSS; fips-186-3, deterministic-rfc6979.
+    :param signature_max_validity: The signature max validity,
+        expressed as a datetime.timedelta value.
+    :param signed_headers: A list of strings. Each value is the name of a HTTP header
+        that must be included in the HTTP signature calculation.
+        The two special signature headers '(request-target)' and '(created)' SHOULD be
+        included in SignedHeaders.
+        The '(created)' header expresses when the signature was created.
+        The '(request-target)' header is a concatenation of the lowercased :method, an
+        ASCII space, and the :path pseudo-headers.
+        When signed_headers is not specified, the client defaults to a single value,
+        '(created)', in the list of HTTP headers.
+        When SignedHeaders contains the 'Digest' value, the client performs the
+        following operations:
+        1. Calculate a digest of request body, as specified in RFC3230, section 4.3.2.
+        2. Set the 'Digest' header in the request body.
+        3. Include the 'Digest' header and value in the HTTP signature.
+
+    :Example:
+
+    Configure API client with HTTP basic authentication:
+      conf = {{{packageName}}}.Configuration(
+          username='the-user',
+          password='the-password',
+      )
+
+    Configure API client with HTTP signature authentication. Use the 'hs2019' signature scheme,
+    sign the HTTP requests with the RSA-SSA-PSS signature algorithm, and set the expiration time
+    of the signature to 5 minutes after the signature has been created.
+      conf = {{{packageName}}}.Configuration(
+        key_id='my-key-id',
+        private_key_path='rsa.pem',
+        signing_scheme='hs2019',
+        signing_algorithm='PSS',
+        signed_headers=['(request-target)', '(created)', 'host', 'date', 'Content-Type', 'Digest']
+        signature_max_validity=timedelta(minutes=5),
+      )
     """
 
     def __init__(self, host="{{{basePath}}}",
                  api_key=None, api_key_prefix=None,
-                 username="", password=""):
+                 username="", password="",
+                 key_id=None, private_key_path=None, signing_scheme=None,
+                 signing_algorithm=None, signature_max_validity=None, signed_headers=None):
```

Comment:
It may be time to separate these into separate dict inputs with None defaults.
We are up to 10 non self arguments.
@sebastien-rosset what do you think?
How about something like:
`(self, host="{{{basePath}}}", api_key_info=None, basic_auth_info=None, signing_info=None)`

Reply:
I think I'll do that for signing info, but I'd rather keep the others as is, I'm concerned the breaking change will cause acceptance issues.

Reply:
Good catch, thanks. I keep thinking that these changes are for python-experimental, but this one is for python + python-experimental. Splitting out signing_info will help us here.
If we want to be aggressive we could make a separate template for python-experimental and switch to using the above 3 dicts. But that's probably best kept as work for another day in a future PR. What do you think?

Reply:
Yes, I was thinking this could be done as a separate PR. It's already big enough, and I am worried it's going to stay in limbo state for a while.